forge-gui/res/cardsfolder/upcoming/smoldering_stagecoach.txt

```diff
@@ -0,0 +1,20 @@
+Name:Smoldering Stagecoach
+ManaCost:3 R
+Types:Artifact Vehicle
+PT:*/5
+S:Mode$ Continuous | EffectZone$ All | CharacteristicDefining$ True | SetPower$ X | Description$ CARDNAME's power is equal to the number of instant and sorcery cards in your graveyard.
+SVar:X:Count$ValidGraveyard Instant.YouOwn,Sorcery.YouOwn
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigInstantEffect | TriggerDescription$ Whenever CARDNAME attacks, the next instant spell and the next sorcery spell you cast this turn each have cascade.
+SVar:TrigInstantEffect:DB$ Effect | StaticAbilities$ CascadeInstant | Triggers$ ExileInstantEff | SubAbility$ DBSorceryEffect
+SVar:DBSorceryEffect:DB$ Effect | StaticAbilities$ CascadeSorcery | Triggers$ ExileSorceryEff
+SVar:CascadeInstant:Mode$ Continuous | EffectZone$ Command | Affected$ Card.Instant+YouCtrl | AffectedZone$ Stack | AddKeyword$ Cascade | Description$ The next instant spell you cast this turn has cascade.
+SVar:CascadeSorcery:Mode$ Continuous | EffectZone$ Command | Affected$ Card.Sorcery+YouCtrl | AffectedZone$ Stack | AddKeyword$ Cascade | Description$ The next sorcery spell you cast this turn has cascade.
+SVar:ExileInstantEff:Mode$ SpellCast | EffectZone$ Command | ValidCard$ Card.Instant+YouCtrl | Execute$ RemoveInstantEff | Static$ True
+SVar:ExileSorceryEff:Mode$ SpellCast | EffectZone$ Command | ValidCard$ Card.Sorcery+YouCtrl | Execute$ RemoveSorceryEff | Static$ True
+SVar:RemoveInstantEff:DB$ ChangeZone | Origin$ Command | Destination$ Exile | Defined$ Self
+SVar:RemoveSorceryEff:DB$ ChangeZone | Origin$ Command | Destination$ Exile | Defined$ Self
```

Comment:
this one is duplicated, you can just have 1 Remove SVar


@tool4ever could we maybe extend the `UntilAPlayerCastSpell` Duration to have the Effect handle itself?

Reply:
hmn I don't see why that shouldn't work already?

Reply:
@tool4ever because it is "Until **A Player** Cast", but we need "Until **You** Cast"

Reply:
oh right, looks like it's only doing some awkward type check

Reply:
implementing a better cleanup might be nice, I just haven't gotten around to it

Reply:
alright, should be better now
You can have the job of cleaning up some others like _Marshland Bloodcaster_ with this if you like :P

Reply:
Thanks and I'll add that to my notes :)